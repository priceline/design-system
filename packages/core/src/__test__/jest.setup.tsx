import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
