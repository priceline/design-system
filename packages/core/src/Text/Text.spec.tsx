import { render, screen } from '../__test__/testing-library'
import { Text } from './Text'

describe('Text', () => {
  test('renders', () => {
    const { asFragment } = render(<Text />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('textAlign prop sets text-align', () => {
    const { asFragment } = render(<Text textAlign='center' />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('regular prop sets font-weight', () => {
    const { asFragment } = render(<Text regular />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('bold prop sets font-weight', () => {
    const { asFragment } = render(<Text bold />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('caps prop sets text-transform', () => {
    const { asFragment } = render(<Text caps />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('fontSize prop sets font-size', () => {
    const { rerender, asFragment } = render(<Text fontSize={0}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Text fontSize={1}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Text fontSize={2}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Text fontSize={3}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Text fontSize={4}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Text fontSize={5}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()

    rerender(<Text fontSize={6}>Text</Text>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('mt prop sets margin-top', () => {
    const { asFragment } = render(<Text mt={2} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('display and overflow', () => {
    const { asFragment } = render(<Text display='inline-block' overflow='none' />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('height and width', () => {
    const { asFragment } = render(<Text height={150} width={200} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('min/max values', () => {
    const { asFragment } = render(<Text minHeight={200} maxHeight={400} minWidth={200} maxWidth={400} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('fontStyle', () => {
    render(<Text fontStyle='italic'>Italic Text</Text>)
    expect(screen.getByText('Italic Text')).toHaveStyleRule('font-style', 'italic')
  })

  test('textDecoration', () => {
    render(<Text textDecoration='underline'>Underlined Text</Text>)
    expect(screen.getByText('Underlined Text')).toHaveStyleRule('text-decoration', 'underline')
  })

  test('textTransform', () => {
    render(<Text textTransform='lowercase'>Lowercase Text</Text>)
    expect(screen.getByText('Lowercase Text')).toHaveStyleRule('text-transform', 'lowercase')
  })
})
