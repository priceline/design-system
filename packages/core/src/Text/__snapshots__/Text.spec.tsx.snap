// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text bold prop sets font-weight 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-weight: 700;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Text caps prop sets text-transform 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  text-transform: uppercase;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Text display and overflow 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: inline-block;
  overflow: none;
}

<div
    class="c0"
  >
    <div
      class="c1"
      display="inline-block"
      overflow="none"
    />
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 12px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="0"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 2`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 14px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="1"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 3`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 16px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="2"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 4`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 20px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="3"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 5`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 24px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="4"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 6`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 32px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="5"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text fontSize prop sets font-size 7`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-size: 40px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      font-size="6"
    >
      Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Text height and width 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  height: 150px;
  width: 200px;
}

<div
    class="c0"
  >
    <div
      class="c1"
      height="150"
      width="200"
    />
  </div>
</DocumentFragment>
`;

exports[`Text min/max values 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  min-height: 200px;
  max-height: 400px;
  min-width: 200px;
  max-width: 400px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Text mt prop sets margin-top 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  margin-top: 8px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Text regular prop sets font-weight 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports[`Text renders 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class=""
    />
  </div>
</DocumentFragment>
`;

exports[`Text textAlign prop sets text-align 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  text-align: center;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
